### PR TITLE
Adds support for single layer CDNs

### PR DESCRIPTION
**Pull request by dg4prez**
#### What does this PR do?
This PR adds support for single layer CDN environments.  Rather than use the server type to determine how the parent data is generated, the code determines if the cache has any parent caches assigned.  If no parent cachegroups are assigned, or if the parent cachegroups are origin cachegroups, parent configuration will be handled as though the cache in question was a mid, or more to the point, the top level of the CDN.

Fixes #(issue_number)
#2133 

#### Which TC components are affected by this PR?

- [ ] Documentation
- [ ] Grove
- [ ] Traffic Analytics
- [ ] Traffic Monitor
- [X] Traffic Ops
- [ ] Traffic Ops ORT
- [ ] Traffic Portal
- [ ] Traffic Router
- [ ] Traffic Stats
- [ ] Traffic Vault
- [ ] Other _________

#### What is the best way to verify this PR?
Compare parent configuration files taken prior to the upgrade to the new versions.  There should be no change for standard multi-tier CDNs.  Single-layer CDNs should show an empty parent.config unless an MSO is present.

#### Check all that apply

- [ ] This PR includes tests
- [ ] This PR includes documentation updates
- [ ] This PR includes an update to CHANGELOG.md
- [ ] This PR includes all required license headers
- [ ] This PR includes a database migration (ensure that migration sequence is correct)
- [ ] This PR fixes a serious security flaw. Read more: [www.apache.org/security](http://www.apache.org/security/)

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2570/
Test PASSed.

**Comment by dg4prez** [comment]
Compared before/after configurations from multiple edge types, mids, and single-layer edges against production database.  No changes in parent configuration files except for single-layer edges, which is the expected behavior.

File: traffic_ops/app/lib/API/Configs/ApacheTrafficServer.pm [hunk 1]
```diff
@@ -2347,6 +2354,33 @@ sub parent_dot_config { #fix qstring - should be ignore for quika
 	my $server_obj = shift;
 	my $data;
 	my $server_type = $server_obj->type->name;
+	
+	## By default we're going to assume that our cache is not a top level cache (mids, edges without parents, etc.)
+	my $is_top_level_cache = 0;
+
+	## Look up parent cachegroup IDs.
+	my $parent_cachegroup_id =
```
[the remark this answers is not in the document]
Comment (dg4prez) [reply traffic_ops/app/lib/API/Configs/ApacheTrafficServer.pm]: Yeah, I was thinking i should do something like that.  Good call.  Thanks!

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2594/
Test FAILed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2595/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2601/
Test FAILed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2602/
Test PASSed.

**Pull request merged**